app/src/main/java/com/kickstarter/viewmodels/BackingAddOnsFragmentViewModel.kt

```diff
@@ -191,27 +191,35 @@ class BackingAddOnsFragmentViewModel {
                     .combineLatest(this.retryButtonPressed.startWith(false), projectAndReward) { _, projectAndReward ->
                         return@combineLatest this.apiClient
                                 .fetchShippingRules(projectAndReward.first, projectAndReward.second)
-                                .doOnError { this.showErrorDialog.onNext(true) }
+                                .doOnError {
+                                    this.showErrorDialog.onNext(true)
+                                    this.shippingSelectorIsGone.onNext(true)
+                    }
                                 .onErrorResumeNext(Observable.empty())
                     }
                     .switchMap { it }
                     .map { it.shippingRules() }
                     .compose(bindToLifecycle())
-                    .subscribe(shippingRules)
+                    .subscribe {
+                        shippingRules.onNext(it)
+                        this.shippingSelectorIsGone.onNext(false)
+                    }
 
             Observable
                     .combineLatest(this.retryButtonPressed.startWith(false), project) { _, pj ->
                         return@combineLatest this.apolloClient
                                 .getProjectAddOns(pj.slug()?.let { it } ?: "")
-                                .doOnError { this.showErrorDialog.onNext(true) }
+                                .doOnError {
+                                    this.showErrorDialog.onNext(true)
+                                    this.shippingSelectorIsGone.onNext(true)}
```

Comment:
No need to update the shippingSelector here

Reply:
removed!

Reply:
Sorry my bad! that is actually correct, we update the visibility of the shipping selector if any of the two network calls fail. You're where doing it correctly ignore my previous comment